### ISSUE TITLE
[inductor][cpu] GPT2ForSequenceClassification AMP max autotune accuracy failure in 2025-08-06 nightly release

### ISSUE THREAD

**Issue by zxd1997066** [pytorch/pytorch#160617]
### 🐛 Describe the bug

the bad commit: b7860c7863df9139bceabf5b0c186e6e4287aa4c
```
/workspace/pytorch# bash inductor_single_run.sh multiple inference accuracy huggingface GPT2ForSequenceClassification amp first static default 0 inductor_max_autotune
Testing with inductor_max_autotune.
multi-threads testing....
/opt/conda/lib/python3.10/site-packages/huggingface_hub/file_download.py:945: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
loading model: 0it [00:01, ?it/s]
cpu  eval  GPT2ForSequenceClassification
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
AUTOTUNE linear_unary(1024x768, 2304x768, 2304)
strides: [768, 1], [1, 0], [1]
dtypes: torch.bfloat16, torch.bfloat16, torch.bfloat16
  cpp_CppMicroGemmAMX_0 0.1922 ms 100.0%
  _linear_pointwise 0.3685 ms 52.2%
SingleProcess AUTOTUNE benchmarking takes 0.3498 seconds and 2.1188 seconds precompiling for 2 choices
AUTOTUNE linear_unary(1024x768, 768x768, 768)
strides: [768, 1], [1, 0], [1]
dtypes: torch.bfloat16, torch.bfloat16, torch.bfloat16
  cpp_CppMicroGemmAMX_1 0.0763 ms 100.0%
  _linear_pointwise 0.2216 ms 34.4%
SingleProcess AUTOTUNE benchmarking takes 0.2990 seconds and 2.1213 seconds precompiling for 2 choices
AUTOTUNE linear_unary(1024x768, 3072x768, 3072)
strides: [768, 1], [1, 0], [1]
dtypes: torch.bfloat16, torch.bfloat16, torch.bfloat16
  cpp_CppMicroGemmAMX_2 0.2373 ms 100.0%
  _linear_pointwise 0.4109 ms 57.7%
SingleProcess AUTOTUNE benchmarking takes 0.3712 seconds and 2.1131 seconds precompiling for 2 choices
AUTOTUNE linear_unary(1024x3072, 768x3072, 768)
strides: [3072, 1], [1, 0], [1]
dtypes: torch.bfloat16, torch.bfloat16, torch.bfloat16
  cpp_CppMicroGemmAMX_3 0.2201 ms 100.0%
  _linear_pointwise 0.4403 ms 50.0%
SingleProcess AUTOTUNE benchmarking takes 0.3719 seconds and 2.1189 seconds precompiling for 2 choices
AUTOTUNE linear_unary(1024x768, 2x768)
strides: [768, 1], [1, 0]
dtypes: torch.bfloat16, torch.bfloat16
  cpp_CppMicroGemmAMX_48 0.0084 ms 100.0%
  _linear_pointwise 0.0611 ms 13.8%
SingleProcess AUTOTUNE benchmarking takes 0.2758 seconds and 2.0711 seconds precompiling for 2 choices
E0814 03:12:24.539000 28184 /workspace/pytorch/torch/_dynamo/utils.py:3032] RMSE (res-fp64): 0.00602, (ref-fp64): 0.00100 and shape=torch.Size([1, 2]). res.dtype: torch.bfloat16, multiplier: 3.000000, tol: 0.005000, use_larger_multiplier_for_smaller_tensor: 0
E0814 03:12:24.539000 28184 /workspace/pytorch/torch/_dynamo/utils.py:2893] Accuracy failed for key name logits
fail_accuracy
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
dev,name,batch_size,accuracy,calls_captured,unique_graphs,graph_breaks,unique_graph_breaks,autograd_captures,autograd_compiles,cudagraph_skips,compilation_latency
cpu,GPT2ForSequenceClassification,1,fail_accuracy,1270,2,0,0,0,0,2,81.739440
```
the last good commit: 9222552572890fd7ff1d75884cc6a99a55ab90bc
```
/workspace/pytorch# bash inductor_single_run.sh multiple inference accuracy huggingface GPT2ForSequenceClassification amp first static default 0 inductor_max_autotune
Testing with inductor_max_autotune.
multi-threads testing....
/opt/conda/lib/python3.10/site-packages/huggingface_hub/file_download.py:945: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
loading model: 0it [00:01, ?it/s]
cpu  eval  GPT2ForSequenceClassification
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
pass
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
dev,name,batch_size,accuracy,calls_captured,unique_graphs,graph_breaks,unique_graph_breaks,autograd_captures,autograd_compiles,cudagraph_skips,compilation_latency
cpu,GPT2ForSequenceClassification,1,pass,1270,2,0,0,0,0,2,69.260649
```
</table>

### Versions

</table><p>SW info</p><table border="1" class="dataframe table">
  <thead>
    <tr style="text-align: right;">
      <th>name</th>
      <th>target_branch</th>
      <th>target_commit</th>
      <th>refer_branch</th>
      <th>refer_commit</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>torchbench</td>
      <td>main</td>
      <td>e03a63be</td>
      <td>main</td>
      <td>fdea2a14</td>
    </tr>
    <tr>
      <td>torch</td>
      <td>main</td>
      <td>efc4b460b3789d97b87d775f86c5f4ee8dfb5629</td>
      <td>main</td>
      <td>f63673626d4</td>
    </tr>
    <tr>
      <td>torchvision</td>
      <td>main</td>
      <td>0.22.0a0+966da7e</td>
      <td>main</td>
      <td>0.22.0a0+966da7e</td>
    </tr>
    <tr>
      <td>torchtext</td>
      <td>main</td>
      <td>0.16.0a0+b0ebddc</td>
      <td>main</td>
      <td>0.16.0a0+b0ebddc</td>
    </tr>
    <tr>
      <td>torchaudio</td>
      <td>main</td>
      <td>2.8.0a0+9b57c7b</td>
      <td>main</td>
      <td>2.8.0a0+f6dfe12</td>
    </tr>
    <tr>
      <td>torchdata</td>
      <td>main</td>
      <td>0.7.0a0+11bb5b8</td>
      <td>main</td>
      <td>0.7.0a0+11bb5b8</td>
    </tr>
    <tr>
      <td>dynamo_benchmarks</td>
      <td>main</td>
      <td>nightly</td>
      <td>main</td>
      <td>nightly</td>
    </tr>
  </tbody>
</table>

Repro:
[inductor_single_run.sh](https://github.com/chuanqi129/inductor-tools/blob//main/scripts/modelbench/inductor_single_run.sh)
bash inductor_single_run.sh multiple inference accuracy huggingface GPT2ForSequenceClassification amp first static default 0 inductor_max_autotune
Suspected guilty commit: https://github.com/pytorch/pytorch/commit/b7860c7863df9139bceabf5b0c186e6e4287aa4c
[huggingface-GPT2ForSequenceClassification-inference-amp-dynamic-default-multiple-accuracy-crash_guilty_commit.log](https://github.com/user-attachments/files/21765696/huggingface-GPT2ForSequenceClassification-inference-amp-dynamic-default-multiple-accuracy-crash_guilty_commit.log)

cc @chuanqi129 @LifengWang
